fix: Datepicker Field: Fix the UX atrocities committed by the payload team

AlessioGr opened the pull request:
## Description

<!-- Please include a summary of the pull request and any related issues it fixes. Please also include relevant motivation and context. -->
Before: Me trying to go to 2017:
![Arc 2023-03-18 at 23 04 12](https://user-images.githubusercontent.com/70709113/226142666-b65010a8-17c6-465c-bf76-8db8135d3c5d.gif)

After:
![Arc 2023-03-18 at 23 11 47](https://user-images.githubusercontent.com/70709113/226142943-5d739c4b-eca5-487c-bd1e-5f6b361856b6.gif)


- [X] I have read and understand the CONTRIBUTING.md document in this repository

## Type of change

<!-- Please delete options that are not relevant. -->

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

I'm actually not sure if this counts as fix or feature.

## Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Existing test suite passes locally with my changes
- [ ] I have made corresponding changes to the documentation

AlessioGr commented:
![Arc 2023-03-18 at 23 12 51@2x](https://user-images.githubusercontent.com/70709113/226142962-b5ca8977-9129-48b3-a60e-0cb794d4fc56.jpg)
Should have showcased it on this field which is closer to what I'm using currently. Because for this one, changing the text doesn't actually update the date

DanRibbens commented on src/admin/components/elements/DatePicker/DatePicker.tsx:
```diff
@@ -49,6 +50,22 @@ const DateTime: React.FC<Props> = (props) => {
     else dateTimeFormat = 'MMM d, yyy';
   }
 
+  const years = Array.from({ length: getYear(new Date()) - 1990 + 1 }, (_, i) => i + 1990);
```
Is this limiting the dropdown to no dates before 1990?
I haven't pulled this down yet.

AlessioGr replied:
Yep! Doesn't affect what dates are possible - just what is shown in the Dropdown.
Should I lower this?

DanRibbens replied:
I pulled it down. The dropdown is very effective for getting to a certain year, as long as it is range. The more concerning thing I am seeing is that using the arrows, I can't choose a date outside of the dropdown year range.

For example if I am at Dec 2023 and click the right arrow, the picker now jumps to Jan 1990.
![image](https://user-images.githubusercontent.com/6434612/226972850-c95e03b6-1904-43ed-ada6-047cb2364019.png)
![image](https://user-images.githubusercontent.com/6434612/226972874-2ca7f411-0091-4a9d-843f-b3c03d6aeb6f.png)

DanRibbens commented:
It would be ideal if the year selector respected the minDate and maxDate options and if there was a way to pick dates outside the year range within the picker.
Others on the team will have better UX feedback than me. I think this is an area we can improve for sure.

AlessioGr commented:
Fixed the arrows, by manually increasing or decreasing the allowed years if the minDate or maxDate is decreased after the arrow is clicked. Check it out again!

JessChowdhury commented:
Hey @AlessioGr - a great feature for the date field would be allowing props to be passed directly to the [react-datepicker](https://reactdatepicker.com/) and that would also give us this functionality.

Then we can simply use `showYearDropdown` and `showMonthDropdown` or similar.
![Screen Shot 2023-03-27 at 5 21 04 PM](https://user-images.githubusercontent.com/67977755/228002643-99070b8e-6a71-4221-8e43-cf19c338c043.png)

AlessioGr commented:
Hey @JessChowdhury, oh my god. I wish I had seen that earlier before doing all that work!!! ahh

It wouldn't be totally useless like that, though. Having it custom implemented like this would allow us to easily replace the normal select with a combobox (like https://headlessui.com/react/combobox) in the future, where you could both select and enter the month.
What do you think?

JarrodMFlesch commented:
I think a lot of this could be done like @JessChowdhury said using the date picker built in props. I envision us removing the left/right arrows in favor of the select fields. I believe all of this could be done with CSS targeting.

I also feel like we should style the select fields (not the options menu) to more fit the feel of the date picker. The goal being to make them "feel" like the header, but with carets to denote that they are selectable.

@AlessioGr do you think you could give this a shot? I know you are wanting to do the combobox, but for now I think we should just use what we already have with the date picker. If we want to revisit this with a different approach later on we can always do that.

AlessioGr commented:
@JarrodMFlesch check it out again! I've changed it to the option already included with react-datepicker:

![Screenshot 2023-07-17 at 23 49 57@2x](https://github.com/payloadcms/payload/assets/70709113/6aeef379-b8b5-47b1-8929-eece892498eb)


Also applied some styling to the select. I hid the original text-only header using a display: none rule, as I found no native option to disable it.

Fixed a couple of styling issues with the monthOnly date picker along the way too.
